projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -256,6 +269,16 @@ export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, O
         this.panels.forEach(panel => panel.collapse());
     }
 
+    private collapseAllExceptLast(): void {
+        const lastExpanded = this.panels?.filter(p => !p.collapsed && !p.header.disabled).pop();
+        this.panels?.forEach((p: IgxExpansionPanelComponent) => {
+            if (p !== lastExpanded && !p.header.disabled) {
+                p.collapsed = true;
+            }
+            this.cdr.detectChanges();
```

Comment:
Here you still run `detectChanges()` on every iteration. Move it outside of the `forEach` loop so it's ran once after all iterations.